www-client/elinks/elinks-9999.ebuild

```diff
@@ -60,7 +64,6 @@ RDEPEND="
 DEPEND="${RDEPEND}
 	X? ( x11-base/xorg-proto )"
 BDEPEND="
-	${PYTHON_DEPS}
```

Comment:
Why was this there before? Does the build system invoke a Python script otherwise?

Reply:
It does, yes, but typically this will trigger the "internal meson sys.executable or `meson runpython`" codepath, which means that depending on dev-build/meson is enough -- not simply because it transitively guarantees python is also installed, but because meson actually exposes its own python.

So I'm unsure whether that's generally an important dependency to keep. I guess I could hear both arguments. There are no non-stdlib python modules imported by said scripts, though.

Reply:
It's also kind of awkward since PYTHON_DEPS is something different with python-single-r1 vs python-any-r1. Reusing PYTHON_DEPS from the "building a scripting extension" for the BDEPEND means that you're now dependent on the value of PYTHON_TARGETS (and setting it to -* will break, even if you disable USE=python).

Reply:
How is this going to work for muon?

Reply:
Would require meson.eclass growing "either-or" muon support which feels a bit unlikely at the moment, e.g. no native file support which is currently relied on.

I have not thus far tried to figure out to what degree muon makes sense in environments where one wishes to *allow the user to choose* which impl, as opposed to environments where some packages explicitly build with muon and some explicitly build with meson (e.g. bootstrappers).

Reply:
OK, fair enough.

Reply:
That's for the documentation as I observed. `www-client/elinks` should add `+docs` (or just -docs which I prefer but maybe would be surprising to other users) already so that can be made optional.

Reply:
Sorry, what are you talking about?

This review comment has no relationship to docs. If you want to make your own review comment about docs, please make your own review comment rather than confusingly responding to an existing unrelated one.

Reply:
`./doc/tools/asciidoc/asciidoc.py` is called at build time and that requires python:3.

Reply:
I simply answered "Why was this there before? Does the build system invoke a Python script otherwise?".

That could be made optional if `docs` is an existing use flag. So instead of plain `${PYTHON_DEPS}`, you can have `docs? ( ${PYTHON_DEPS} )`.  And then you add `$(meson_use docs)` to emesonargs.

Reply:
I already answered that question before you replied, though?

And no, this isn't really related to docs (which use doxygen) since the asciidoc stuff is just automatically built. Also, there's another python script in a different directory, so python is necessary even without the docs/ directory in the first place.

Also this doesn't answer my original rationale for leaving it out, so I'm not sure why you're mentioning it -- given that regardless of whether this dependency is kept in BDEPEND or removed, the asciidoc targets will always build and install, so you're reviewing the wrong lines and being extremely confusing.

Reply:
> I already answered that question before you replied, though?

Yes but you only mentioned meson related stuff which even seems to make you unsure whether that's generally an important dependency to keep.

So I decided to reply with what I think is a definite reason why it should be.

And as I just searched now, `${PYTHON_DEPS}` appears to be already a dependency before meson support so whatever reason it was added, it's not because of meson: https://github.com/gentoo/gentoo/commit/51da4a652f7f31e23466bc7b5e6e2a54ff534331

Python was also explicitly required as a dependency in `configure.in` when upstream bundled `asciidoc.py`: https://github.com/rkd77/elinks/commit/8e20417bdb20f01479fe8d93753985b1c9a14f83

> And no, this isn't really related to docs (which use doxygen) since the asciidoc stuff is just automatically built.

Asciidoc stuff aren't built automatically if `doc` is disabled through meson.  I think it's officially considered as a "docs" as well even though they're not made through doxygen.

> Also, there's another python script in a different directory, so python is necessary even without the docs/ directory in the first place.

I only saw contrib scripts and stuff related to msys.

> Also this doesn't answer my original rationale for leaving it out, so I'm not sure why you're mentioning it -- given that regardless of whether this dependency is kept in BDEPEND or removed.

As I said above.

> the asciidoc targets will always build and install, so you're reviewing the wrong lines and being extremely confusing.

I was merely giving a suggestion to make it optional so in some sense it can be "left out".  What's even better is that you can explicitly show why the dependency is needed.  And that's through `doc? ( ${PYTHON_DEPS} )`.  In a way it's still on topic on how `${PYTHON_DEPS}` should be treated -- that which seems to be the real intent of this discussion, no?